features/device_key/source/DeviceKey.cpp

```diff
@@ -260,7 +260,6 @@ int DeviceKey::generate_key_by_random(uint32_t *output, size_t size)
         return DEVICEKEY_INVALID_PARAM;
     }
 
-#if DEVICE_TRNG
```

Comment:
This section of code is only safe if either the device has a TRNG, or MBEDTLS_ENTROPY_NV_SEED is turned on. The `mbedtls_entropy_func()` might catch it if this is not the case and return an error, but it depends on the Mbed TLS configuration. I think it would be safer to add `&& defined(MBEDTLS_ENTROPY_NV_SEED)` to the condition than remove the macro completely. Would that solve the problem?

Reply:
@yanesca What will happen if there is no entropy seed? Will it be better || defined(MBEDTLS_ENTROPY_NV_SEED)?

Reply:
If a component registers its own weak entropy sources as strong or enables the HAVEGE source, then this function returns low entropy keys. Yes, I think that would be better: it would make explicit that we only provide output if TRNG or entropy seed is present.

Reply:
@yanesca thanks.
I have fixed it as you suggested.